astropy/timeseries/tests/test_downsample.py

```diff
@@ -39,6 +43,30 @@ def test_reduceat():
     )
 
 
+def test_nanmean_reduceat():
+    data = np.arange(8)
+    indices = [0, 4, 1, 5, 5, 2, 6, 6, 3, 7]
+
+    reduceat_output1 = reduceat(data, indices, np.nanmean)
+    nanmean_output1 = nanmean_reduceat(data, indices)
+    assert_equal(reduceat_output1, nanmean_output1)
+
+    data = data.astype("float")
+    data[::2] = np.nan
+    with np.testing.suppress_warnings() as sup:
```

Comment:
Total nitpick, but I'd tend to use `pytest`s machinery, testing that the warning actually gets emitted - then, if `numpy` ever changes, we'll know. And it is a line shorter...
```
with pytest.warns(RuntimeError, match="Mean of empty slice"):
    reduceat_output2 = ...
```
(and same below).

Reply:
I'd argue we should not make our tests sensitive to implementation details outside of our control, so I'm -1 on this suggestion.

Reply:
I'm definitely fine with keeping as is too.